File: 1.Two Sum/TwoSum.cpp
class SolutionBruteForce {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for(int i = 0 ; i < nums.size(); i++)
        {
            for(int j = 1; j < nums.size(); j++)
            {
                if (nums[i]+nums[j]==target)
                    return {i,j};
            }
        }
        return {};
    }
};

class SolutionTwoPass{
    public:
    vector<int>twoSum(vector<int>& nums, int target)
{
    unordered_map<int, int> hash;
    for(int i = 0; i< nums.size(); i++)
    {
        hash[nums[i]] = i;
    }

    for(int i = 0; i< nums.size(); i++)
    {
        int comp = target - nums[i];
        if (hash.find(comp) != hash.end() && hash[comp] != i)
        {
            return {i, hash[comp]};
        }
    }
    return {};
}
};

class Solution{
    public:
    vector<int>twoSum(vector<int>& nums, int target)
{
    unordered_map<int, int> hash;
    for(int i = 0; i< nums.size(); i++)
    {
        int comp = target - nums[i];

        if( hash.find(comp) != hash.end())
        {
            return {i, hash[comp]};
        }

        hash[nums[i]] = i;
    }

    return {};
}
};

File: 1.Two Sum/TwoSum.py
class SolutionBF(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range (len(nums)):
            for j in range (1, len(nums)):
                if nums[j] ==  target - nums[i]:
                    return [i,j]
        return []

class SolutionTwoPass(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash = {}
        for i in range (len(nums)):
            hash[nums[i]] = i
        for j in range (len(nums)):
            comp = target - nums[j]
            if comp in hash and hash[comp]!=j:
                return [j,hash[comp]]
        return []

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash = {}
        for i in range (len(nums)):
            comp = target - nums[i]
            if comp in hash and hash[comp]!=i:
                return [i,hash[comp]]
            hash[nums[i]] = i
        return []
